implementations/rust/ockam/ockam_api/src/nodes/service/portals.rs

```diff
@@ -205,3 +280,78 @@ impl NodeManagerWorker {
         })
     }
 }
+
+/// Configure the session for automatic recovery.
+fn replacer(
+    manager: Arc<RwLock<NodeManager>>,
+    data: Data,
+    bind: String,
+    addr: MultiAddr,
+    auth: Option<IdentityIdentifier>,
+    access: Arc<dyn AccessControl>,
+) -> Replacer {
+    Box::new(move |prev| {
+        let addr = addr.clone();
+        let auth = auth.clone();
+        let bind = bind.clone();
+        let manager = manager.clone();
+        let access = access.clone();
+        let data = data.clone();
+        Box::pin(async move {
+            debug!(%prev, %addr, "creating new tcp inlet");
+            let f = async {
+                let prev = try_multiaddr_to_addr(&prev)?;
+                let mut this = manager.write().await;
+                let _ = this.delete_secure_channel(&prev).await;
+                if let Some(a) = data.get::<MultiAddr>(OUTER_CHAN) {
+                    let a = try_multiaddr_to_addr(&a)?;
+                    let _ = this.delete_secure_channel(&a).await;
+                }
+                let rest = {
+                    let (sec1, rest) = this.connect(&addr, auth).await?;
+                    if !sec1.is_empty()
+                        && rest.matches(0, &[Service::CODE.into(), Secure::CODE.into()])
+                    {
+                        data.put(OUTER_CHAN, sec1.clone());
+                        let addr = sec1.clone().try_with(rest.iter().take(2))?;
+                        let (sec2, _) = this.connect(&addr, None).await?;
+                        sec2.try_with(rest.iter().skip(2))?
+                    } else {
+                        sec1.try_with(&rest)?
+                    }
+                };
+                let r = multiaddr_to_route(&rest)
+                    .ok_or_else(|| ApiError::message(format!("invalid multiaddr: {rest}")))?;
+                if let Some(wa) = data.get::<Address>(INLET_WORKER) {
+                    let _ = this.tcp_transport.stop_inlet(wa).await;
+                }
+                let opts = InletOptions::new(bind, r, access);
+                let wa = this.tcp_transport.create_inlet_extended(opts).await?.0;
+                data.put(INLET_WORKER, wa);
```

Comment:
IIUC, in this block you stop the `INLET_WORKER` if it's set in `data` before adding the new worker address (`wa`). Is that right? My main concern is that I wouldn't feel confident debugging/refactoring/modifying this code without your help.

I know it's not a complex piece of code, but it's hard to contextualize it within all that's going on in the whole function. My problem in this case is that I have to read the entire function to understand what's going on in this specific block and figure out how it depends on the previous code. In my experience, it helps a lot to add a simple visual separation with a small comment so people that don't have the context of this code can quickly navigate through it. 

For example:

```rust
let rest = ...;

// Replace INLET_WORKER with a new worker address
let r = ...
...
data.put(...);
```

This way I could easily identify what the next lines of code are going to do without having to read the code first. Let me know what you think, just trying to make things easier for our future selves!

Reply:
Thanks @adrianbenavides! I added more comments to provide context. Let me know if it helps.

Reply:
It looks great now @twittner, thank you so much for adding those comments :pray: